library/lib_window_menu.ks

```diff
@@ -0,0 +1,87 @@
+// This script shows a menu on the terminal allowing user to
+// select one of the options and return it to calling script.
+@lazyglobal off.
+
+run lib_window.
+
+function open_window_menu{
+	parameter title.
+	parameter list_of_names.
+	parameter window.
+
+	local x is window[0].
+	local y is window[1].
+	
+	local current_option is 0.
+	local len is list_of_names:length().
+
+	local last_up is ag7.
+	local last_down is ag8.
+	local last_sel is ag9.
+	local process_state is list(
+		list(false,window,"update_window_menu",false),
+		current_option,last_up,last_down,last_sel,list_of_names,title
```

Comment:
So what would be passed to the update function in your model? The whole process (list of (1) list of custom fields and (2) system fields), or just the custom fields? In the latter case, it would be impossible to extract window, and the former is no better than my model - it's virtually identical, other than starting indexing from 1, and not 0.